tests/test_util.py

```diff
@@ -1,15 +1,31 @@
 # Copyright 2023 MosaicML Streaming authors
 # SPDX-License-Identifier: Apache-2.0
 
+import json
+import os
+import tempfile
+import time
+import urllib.parse
 from multiprocessing.shared_memory import SharedMemory as BuiltinSharedMemory
-from typing import List, Optional
+from typing import List, Optional, Tuple, Union
 
 import pytest
 
 from streaming.base.constant import RESUME
 from streaming.base.shared.prefix import _get_path
+from streaming.base.storage.download import download_file
+from streaming.base.storage.upload import CloudUploader
 from streaming.base.util import (bytes_to_int, clean_stale_shared_memory, get_list_arg,
-                                 number_abbrev_to_int, retry)
+                                 merge_index, number_abbrev_to_int, retry)
+
+MY_PREFIX = 'train_' + str(time.time())
+MY_BUCKET = {
```

Comment:
Can we have this as `@pytest.mark.parametrize`? I think only `s3://` is getting used.